### DISCUSSION TITLE
Is it possible to stop and restart the ebiten application?

### DISCUSSION THREAD

**Discussion by J7mbo** [hajimehoshi/ebiten#2136] [Q&A]
A weird one, but humour me please! :) For me, Ebiten is not the main application - it's a sub-application within my application that should be stopped and started again when the user does something.

I notice that I cannot call `RunGame()` twice because of a global variable (`isRunGameEnded_`) preventing this. Even after finishing the run loop by returning an `error`, and even with manually calling `defer atomic.StoreInt32(&isRunGameEnded_, 0)` after this in a fork, it doesn't work.

Does anyone know of a way that I could to stop `ebiten`, and then continue running it at a later date? Doesn't have to go into the main repo, I'm okay with keeping this 'weird' one in a fork.

**Comment by hajimehoshi** [comment]
I'd run an Ebitengine application as a different process with os/exec.

**Reply by J7mbo** [reply]
Wouldn't this mean that going to prod with it would require two binaries?

There must be a way to reset the state of the app back to its initial state. From my understanding there's only this var and a thread unlock function...

**Reply by hajimehoshi** [reply]
> Wouldn't this mean that going to prod with it would require two binaries?

Yes.

> There must be a way to reset the state of the app back to its initial state. From my understanding there's only this var and a thread unlock function...

Terminating the graphics state correctly for restarting is not obvious and pretty hard to implement unfortunately (at least in the Ebitengine implementation).
